components/admin_console/billing/billing_subscriptions/limits.tsx

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {cloudFreeEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getCloudLimits} from 'mattermost-redux/selectors/entities/cloud';
+import {getCloudLimits as getCloudLimitsAction} from 'actions/cloud';
+
+import LimitCard from './limit_card';
+
+export const GB = 1 * 8 * 1024 * 1024 * 1024;
+
+// TODO: Replace this with actual usages stored in redux,
+// that ideally are updated with a websocket event in near real time.
+const fakeUsage = {
+    files: {
+        totalStorage: Number(GB),
+    },
+    messages: {
+        history: 11000,
+    },
+    boards: {
+        cards: 300,
+    },
+    integrations: {
+        enabled: 9,
+    },
+};
+
+const Limits = (): JSX.Element | null => {
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+    const cloudLimits = useSelector(getCloudLimits);
+    const dispatch = useDispatch();
+    const [requestedLimits, setRequestedLimits] = useState(false);
+
+    useEffect(() => {
+        if (isCloudFreeEnabled && !requestedLimits) {
+            dispatch(getCloudLimitsAction());
+            setRequestedLimits(true);
+        }
+    }, [isCloudFreeEnabled, requestedLimits]);
+
+    if (!isCloudFreeEnabled || !(cloudLimits?.messages?.history)) {
+        return null;
+    }
+
+    return (
+        <div className='ProductLimitsPanel'>
+            <div className='ProductLimitsPanel__title'>
+                <FormattedMessage
+                    id='workspace_limits.upgrade'
+                    defaultMessage='Upgrade to avoid {planName} data limits'
+                    values={{
+                        planName: 'Cloud Starter',
```

Comment:
Should this be localized as well?

Reply:
My hunch is no, but @cwarnermm may know better. Reasons I think no:
* We don't provide localized plan names on our marketing site, so it may not increase the user's understanding if the plan name here is disconnected from the marketing site.
* the plan name is related to the license which is never translated.
* Although we do define `"admin.billing.subscription.planDetails.productName.cloudStarter":`, mostly it isn't translated. If we change the plan name, we now force all the translators to retranslate this string, which for most languages results in busy work of copy-pasting the english text into their language they are translating.

```
i18n/en.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/sv.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/ja.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/en_AU.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/bg.json
327:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/es.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/pl.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/zh-CN.json
327:  "admin.billing.subscription.planDetails.productName.cloudStarter": "入门级云",

i18n/fr.json
332:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/hu.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/ru.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/tr.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",

i18n/de.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter Edition",

i18n/nl.json
334:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Wolk Starter",

i18n/fa.json
315:  "admin.billing.subscription.planDetails.productName.cloudStarter": "Cloud Starter",
``